File: 300/334 - Increasing Triplet Subsequence.c
#include <limits.h>

bool increasingTriplet(int* nums, int numsSize) {
    int a = INT_MIN, b = INT_MIN, i;
    
    for (i = 0; i < numsSize; ++i) {
        if (a == INT_MIN || a >= nums[i])
            a = nums[i];
        else if (b == INT_MIN || b >= nums[i])
            b = nums[i];
        else
            return 1;
    }
    
    return 0;
}

File: csharp/334.increasing-triplet-subsequence.cs
/*
 * @lc app=leetcode id=334 lang=csharp
 *
 * [334] Increasing Triplet Subsequence
 */
public class Solution {
    public bool IncreasingTriplet(int[] nums) {
        int first = int.MaxValue, second = int.MaxValue;
        foreach (int num in nums) {
            if (first >= num) {
                first = num;
            } else if (second >= num) {
                second = num;
            } else {
                return true;
            }
        }
        return false;
    }
}

